nannou/src/draw/mesh/mod.rs

```diff
@@ -181,75 +176,72 @@ where
     }
 }
 
-impl<S> Default for Mesh<S> {
+impl Default for Mesh {
     fn default() -> Self {
         let mesh = Default::default();
         Mesh { mesh }
     }
 }
 
-impl<S> Deref for Mesh<S> {
-    type Target = MeshType<S>;
+impl Deref for Mesh {
+    type Target = MeshType;
     fn deref(&self) -> &Self::Target {
         &self.mesh
     }
 }
 
-impl<S> DerefMut for Mesh<S> {
+impl DerefMut for Mesh {
     fn deref_mut(&mut self) -> &mut Self::Target {
         &mut self.mesh
     }
 }
 
-impl<S> mesh::GetVertex<u32> for Mesh<S>
-where
-    S: Clone,
-{
-    type Vertex = Vertex<S>;
+impl mesh::GetVertex<u32> for Mesh {
+    type Vertex = Vertex;
```

Comment:
Unnecessary

Reply:
Thanks for taking a look at the PR!

These `type A = B;` lines are not normal type aliases. They are "associated types" of the trait within which they're nested. So despite it looking like a type alias that says `type Foo = Foo;` it's actually part of the implementation of the trait.

In this case it says "the associated type `Vertex` of the implementation of `GetVertex<u32>` for `Mesh` is the type `Vertex`", where the trait's associated type `Vertex` just happens to share the same name as `Vertex` type from the outer scope.

The trait having an associated type like this is useful when writing generic code. By saying `T: GetVertex` in a generic type or function, you can then also talk about `T::Vertex` without having to declare a new generic type parameter for the vertex type. I think the `Iterator` trait's associated type `Item` is probably the most prevalent example.

Going to resolve these, but let me know if you meant something else and I'm missing something!

Reply:
Thanks for the explanation, I should have researched it more! I was thrown off by the X = X and initially thought it was a potential "replace all" situation.